Utils/src/FastSimWeightPR31285To36122.cc

```diff
@@ -0,0 +1,246 @@
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(19,6.516129);
+   hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5->SetBinContent(20,7.685393);
+   
+   
+}
+
+FastSimWeightPR31285To36122::~FastSimWeightPR31285To36122() {
+  // do anything here that needs to be done at destruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+
+// ------------ method called for each event  ------------
+void FastSimWeightPR31285To36122::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+    using namespace edm;
+
+    double EventWeight = 1.0;    
+    
+    edm::Handle< View<reco::GenParticle> > genPartCands;
+    iEvent.getByToken(genCollectionTok, genPartCands);
+    
+    edm::Handle< View<reco::GenJet> > genJets;
+    iEvent.getByToken(genJetTok, genJets);
+    
+    edm::Handle< View<pat::Jet> > recJets;
+    iEvent.getByToken(recJetTok, recJets);   
+    
+    
+    bool jetHasOffendingGp_;
+    TLorentzVector genJet_tlv;
+    TLorentzVector recJet_tlv;
+    TLorentzVector genPart_tlv;    
+    float dr;    
+    float mindr;
+    double leadHadronPt;
+    float originxy;   
+    float decayxy;       
+    int hadronFlavor;     
+    int pdgid;
+    
+
+    for(const auto& genJet : *genJets) 
+    {
+           if (!(genJet.pt()>30)) continue;
+           genJet_tlv.SetPxPyPzE(genJet.px(), genJet.py(), genJet.pz(), genJet.energy());
+           jetHasOffendingGp_ = false;//must use
+           mindr = 99;
+           hadronFlavor = -11;
+           leadHadronPt = 1;
+           originxy = -1;
+           for(const auto& recJet : *recJets) 
+            {
+                recJet_tlv.SetPxPyPzE(recJet.px(), recJet.py(), recJet.pz(), recJet.energy());
+                dr = recJet_tlv.DeltaR(genJet_tlv);
+                if (!(dr<0.4 && dr<mindr)) continue;
+                mindr = dr;
+                hadronFlavor = std::max(1,recJet.hadronFlavour());
+                if (hadronFlavor==22) hadronFlavor = 1;
+                
+                for(const auto& iPart : *genPartCands) {
+                    if (!(iPart.pt()>15)) continue;
+                    pdgid = abs(iPart.pdgId());
+                    genPart_tlv.SetPxPyPzE(iPart.px(), iPart.py(), iPart.pz(), iPart.energy());
+                    if (!(genPart_tlv.DeltaR(genJet_tlv)<0.4)) continue;
+                    if(!(pdgid>100 && pdgid<600)) continue;
+                    if (hadronFlavor<4 && pdgid>100 && pdgid<400)
+                    {
+                        if(iPart.pt()>leadHadronPt) {leadHadronPt = iPart.pt();}
+                    }
+                    else if (hadronFlavor==4 && pdgid>400 && pdgid<500)
+                    {
+                        if(iPart.pt()>leadHadronPt) {leadHadronPt = iPart.pt();}
+                    }
+                    else if (hadronFlavor==5 && pdgid>400 && pdgid<600)
+                    {
+                    if(iPart.pt()>leadHadronPt) {leadHadronPt = iPart.pt();}
+                    }        
```

Comment:
The logic of this block can be simplified to avoid repeating the same condition:
```cpp
if (iPart.pt()>leadHadronPt) {
    if ( (hadronFlavor<4 && pdgid>100 && pdgid<400) or
         (hadronFlavor==4 && pdgid>400 && pdgid<500) or
         (hadronFlavor==5 && pdgid>400 && pdgid<600) )
    {
        leadHadronPt = iPart.pt();
    }
}
```

Reply:
thanks, that's better